scripts/spcgeonode/geoserver/docker-entrypoint.sh

```diff
@@ -132,10 +138,20 @@ echo "6. Importing SSL certificate (if using HTTPS)"
 
 # https://docs.geoserver.org/stable/en/user/community/oauth2/index.html#ssl-trusted-certificates
 if [ ! -z "$HTTPS_HOST" ]; then
+    # Random password are generated every container restart
     PASSWORD=$(openssl rand -base64 18)
-
-    openssl s_client -connect ${HTTPS_HOST#https://}:${HTTPS_PORT} </dev/null |
-        openssl x509 -out server.crt
+    # Since keystore password are autogenerated every container restart,
+    # The same keystore will not be able to be opened again.
+    # So, we create a new one.
+    rm -f /keystore.jks
+
+    # Support for Kubernetes/Docker file secrets if the certificate file path is defined
+    if [ ! -z "${SSL_CERT_FILE}" ]; then
+      cp -f ${SSL_CERT_FILE} server.crt
+    else
+      openssl s_client -connect ${HTTPS_HOST#https://}:${HTTPS_PORT} </dev/null |
+          openssl x509 -out server.crt
+    fi
```

Comment:
Possible bug:
When the container restarts, PASSWORD will be different so the keystore can't be opened by GeoServer.

Addition:
Allow to copy server certificate instead of fetching it using openssl if SSL_CERT_FILE location is defined to know where the certificate is mounted in the pods.